spec/service_worker/index.html

```diff
@@ -1478,6 +1485,59 @@ <h2 id="status">Status of This Document</h2>
     </spec-section>
   </spec-section>
 
+  <spec-section id="install-event-section">
+    <h1><code>InstallEvent</code></h1>
+
+<spec-idl>[Constructor(DOMString <var>type</var>, optional <a href="#extendable-event-init-dictionary">ExtendableEventInit</a> <var>eventInitDict</var>), Exposed=ServiceWorker]
+interface <dfn id="install-event-interface" title="InstallEvent">InstallEvent</dfn> : <a href="#extendable-event-interface">ExtendableEvent</a> {
+  void <a href="#install-event-register-foreign-fetch-scopes-method">registerForeignFetchScopes</a>(sequence&lt;<a href="http://heycam.github.io/webidl/#idl-USVString">USVString</a>&gt; <var>subScopes</var>, (<a href="http://heycam.github.io/webidl/#idl-USVString">USVString</a> or sequence&lt;<a href="http://heycam.github.io/webidl/#idl-USVString">USVString</a>&gt;) <var>origins</var>);
+};
+</spec-idl>
+
+    <spec-section id="register-foreign-fetch-scopes-method">
+      <h1><code>event.registerForeignFetchScopes(<var>subScopes</var>, <var>origins</var>)</code></h1>
+
+      <p><code><a href="#install-event-register-foreign-fetch-scopes-method">registerForeignFetchScopes(<var>subScopes</var>, <var>origins</var>)</a></code> registers this service worker to handle foreign fetches from certain origins for certain sub scopes.</p>
+
+      <p><dfn id="install-event-register-foreign-fetch-scopes-method"><code>registerForeignFetchScopes(<var>subScopes</var>, <var>origins</var>)</code></dfn> method <em class="rfc2119" title="MUST">must</em> run these steps or their <a href="#dfn-processing-equivalence">equivalent</a>:</p>
+      <spec-algorithm>
+      <ol>
+        <li>If the <a href="https://dom.spec.whatwg.org/#dispatch-flag">dispatch flag</a> is unset, then:
+          <ol>
+            <li><a href="http://heycam.github.io/webidl/#dfn-throw">Throw</a> an "<code><a href="http://heycam.github.io/webidl/#invalidstateerror">InvalidStateError</a></code>" exception.</li>
+            <li>Abort these steps.</li>
+          </ol>
+        </li>
+        <li>If <var>origins</var> is empty <a href="http://heycam.github.io/webidl/#dfn-throw">throw</a> a <code>TypeError</code> and abort these steps.</li>
+        <li>Let <var>originURLs</var> be an empty list of <a href="https://url.spec.whatwg.org/#concept-url">URLs</a>.</li>
+        <li>If the value of <var>origins</var> is not a single string equal to a single U+002A ASTERISK character (*):
```

Comment:
> When you already have two arguments and will likely add more in the future it seems better to just have a dictionary from the start since the eventual code will become more readable.

Okay, changed it to a dictionary.